pkg/utils/addons/addons.go

```diff
@@ -1,22 +1,26 @@
 package addons
 
 import (
+	"path"
+
 	mfc "github.com/manifestival/controller-runtime-client"
 	mf "github.com/manifestival/manifestival"
 	"github.com/tektoncd/operator/pkg/flag"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const resourced bool = true
+
+type generateDeployTask func(map[string]interface{}) map[string]interface{}
 type taskGenerator interface {
 	generate(pipeline unstructured.Unstructured) (unstructured.Unstructured, error)
 }
 
-type generateDeployTask func(interface{}) map[string]interface{}
-
 type pipeline struct {
-	environment string
-	nameSuffix  string
+	environment     string
+	nameSuffix      string
+	resourcedSuffix string
 	generateDeployTask
```

Comment:
could you add short comments for each of these fields. It could appear confusing to someone who wants to work on it in the future. I mean, it will be fantastic if someone can make sense of what each field is without having to read the rest of the code.

Reply:
`resourcedSuffix string` 
ok, after reading `createPipelines function, i think, we can handle this better.
what is the real purpose of this.

i see that there is a boolean parameter in
 `func generateBasePipeline(template mf.Manifest, taskGenerators []taskGenerator, resourced bool) ([]unstructured.Unstructured, error) {` 

and later
```
if resourced {
			taskName += "-pr"
		}
```

One way to handle this better is to add a function type, `type pipelineModifier func(map[string]interface{}) map[string]interface{}`

then, may be make

```
type pipeline struct {
	environment     string
	nameSuffix      string
	modifiers []pipelineModifier
```

Then generateDeployTask implemention(s) could be 1 item and for pipelineResource based ones there can be an additional modifiers

eg: 
```
workspacedTaskGenerators := []taskGenerator{
		&pipeline{environment: "openshift", nameSuffix: "", modifiers: []pipelineModifier{modiopenshiftDeployTask}},
		...
	}
```

and 
```
resourcedTaskGenerators := []taskGenerator{
		&pipeline{environment: "openshift", nameSuffix: "",modifiers: []pipelineModifier{modiopenshiftDeployTask, pipelineResourceModifications} openshiftDeployTask},
...	}
```